app.py

```diff
@@ -0,0 +1,305 @@
+def create_primary_distributor():
+    """A function to take user input and creates the Distributor."""
+    locations_to_include = input('Please enter the list(comma seperated) locations to include in \'-\' seperated format\n'
+                                 'Example: PUNCH-JK-IN,TN-IN,SUTAC-JUN-PE,SORAS-AYA-PE\n'
+                                 '++++++++++++++++++++++++++++++++++++++++++++++++++++\n')
+    locations_to_exclude = input('Please enter the list(comma seperated) locations to exclude in \'-\' seperated format\n')
+    invalid_code = check_code_validation(locations_to_include, locations_to_exclude)
+    if invalid_code:
+        print('%s code is invalid, no city exists with that data' %invalid_code)
+        return
+    permissions = {'include' : locations_to_include.split(',') if locations_to_include != '' else [],
+                   'exclude': locations_to_exclude.split(',') if locations_to_exclude != '' else []}
+    distributor = Distributor(permissions)
+    _distributors.append(distributor)
+    print('Distributor created successfully with id: %d and the following permissions' %(distributor.id))
+    print(distributor.permissions_hash)
+
+
+def create_sub_distributor():
+    """A function to take user input and creates the sub Distributor."""
+    parent_distributor_id = input('Enter the parent distributor id:\n')
+    try:
+        parent_distributor = _distributors[int(parent_distributor_id)]
+    except IndexError:
+        print('Sorry, No distributor exists with that id.')
+        return
+    except ValueError:
+        print('Sorry, Invalid ID type.')
+        return
+    locations_to_include = input('Please enter the list(comma seperated) locations to include in \'-\' seperated format\n'
+                                 'Example: PUNCH-JK-IN,TN-IN,SUTAC-JUN-PE,SORAS-AYA-PE\n'
+                                 '++++++++++++++++++++++++++++++++++++++++++++++++++++\n')
+    locations_to_exclude = input('Please enter the list(comma seperated) locations to exclude in \'-\' seperated format\n')
+    invalid_code = check_code_validation(locations_to_include, locations_to_exclude)
+    if invalid_code:
+        print('%s code is invalid, no city exists with that data.' %invalid_code)
+        return
+    locations_to_include_s = []
+    for i in locations_to_include.split(','):
+        if not parent_distributor.has_permission(i):
+            print('Sorry the distributor dont have permissions to assign this location: \t' + i)
+            return
+        else:
+            locations_to_include_s.append(i)
+
+    permissions = {'include' : locations_to_include_s, 'exclude': locations_to_exclude.split(',') if locations_to_exclude != '' else []}
+    distributor = Distributor(permissions, parent_distributor)
+    _distributors.append(distributor)
+    print('Distributor created successfully with the id: %d and following permissions' %(distributor.id))
+    print(distributor.permissions_hash)
+
+def check_the_permission():
```

Comment:
Where's the condition to handle a distributor can authorize/not permission to another distributor ?

Reply:
This logic is written in a method of Distributor class.
```suggestion
def has_permission(self, city_code):
        """Checks the distributor has the permission for the code passed."""
      ```